main.go

```diff
@@ -258,5 +387,23 @@ func main() {
 	// 	logrus.Infof("pmm-agent registered: %s.", cfg.UUID)
 	// }
 
-	workLoop(ctx, cfg, l, client)
+	var wg sync.WaitGroup
+
+	localServer := agentlocal.NewAgentLocalServer(cfg)
+
+	wg.Add(1)
+	go func() {
+		wg.Done()
+		runGRPCServer(ctx, cfg, localServer, grpcAddr)
+	}()
+
+	wg.Add(1)
+	go func() {
+		wg.Done()
+		runJSONServer(ctx, cfg, grpcAddr)
+	}()
+
+	wg.Wait()
+
+	workLoop(ctx, cfg, l, client, localServer)
```

Comment:
Wait. How that can work? We start `workLoop` only after gRPC Server is done?
Does it actually work?

Reply:
Yes, it works. I always run the code and test it. Look at the code above. `wg.Done()` called after goroutine is run not after gRPC server is done.

Reply:
That is super weird and unexpected. What is the point of doing that? Why not just `go runGRPCServer(ctx, cfg, localServer, grpcAddr)`? `WaitGroup` is used for waiting and synchronisation, and there is no waiting of synchronisation.